packages/hash/frontend/src/pages/[account-slug]/entities/[entity-id].page/property-table/extract-enriched-property-types-from-entity.ts

```diff
@@ -0,0 +1,47 @@
+import { PropertyType } from "@blockprotocol/type-system-web";
+import { capitalize } from "@mui/material";
+import { EntityResponse } from "../../../../../components/hooks/blockProtocolFunctions/knowledge/knowledge-shim";
+import { EnrichedPropertyType } from "./types";
+
+const getDataTypesOfPropertyType = (
+  propertyType: PropertyType,
+  entity: EntityResponse,
+) => {
+  /** @todo check why propertyValue does not have with a proper type  */
```

Comment:
Any ideas on this?

Reply:
Updated it, it was a union type, so I should've used property key checking to determine which type it actually is.